Mathlib/Topology/ContinuousFunction/Bounded.lean

```diff
@@ -0,0 +1,1627 @@
+      · exact (hU x').2.2 _ hx' _ (hU x').1 hg
+      · have F_f_g : F (f x') = F (g x') :=
+          (congr_arg (fun f : tα → tβ => (f ⟨x', x'tα⟩ : β)) f_eq_g : _)
+        calc
+          dist (f x') (g x') ≤ dist (f x') (F (f x')) + dist (g x') (F (f x')) :=
+            dist_triangle_right _ _ _
+          _ = dist (f x') (F (f x')) + dist (g x') (F (g x')) := by rw [F_f_g]
+          _ < ε₂ + ε₂ := (add_lt_add (hF (f x')).2 (hF (g x')).2)
+          _ = ε₁ / 2 := add_halves _
+    _ = ε₁ := by rw [add_halves, add_halves]
+#align bounded_continuous_function.arzela_ascoli₁ BoundedContinuousFunction.arzela_ascoli₁
+
+/-- Second version, with pointwise equicontinuity and range in a compact subset. -/
+theorem arzela_ascoli₂ (s : Set β) (hs : IsCompact s) (A : Set (α →ᵇ β)) (closed : IsClosed A)
+    (in_s : ∀ (f : α →ᵇ β) (x : α), f ∈ A → f x ∈ s) (H : Equicontinuous ((↑) : A → α → β)) :
+    IsCompact A := by
+  /- This version is deduced from the previous one by restricting to the compact type in the target,
+  using compactness there and then lifting everything to the original space. -/
+  have M : LipschitzWith 1 (↑) := LipschitzWith.subtype_val s
+  let F : (α →ᵇ s) → α →ᵇ β := comp (↑) M
+  refine' isCompact_of_isClosed_subset ((_ : IsCompact (F ⁻¹' A)).image (continuous_comp M)) closed
+      fun f hf => _
+  · haveI : CompactSpace s := isCompact_iff_compactSpace.1 hs
+    refine' arzela_ascoli₁ _ (continuous_iff_isClosed.1 (continuous_comp M) _ closed) _
+    rw [uniformEmbedding_subtype_val.toUniformInducing.equicontinuous_iff]
+    exact H.comp (A.restrictPreimage F)
+  · let g := codRestrict s f fun x => in_s f x hf
+    rw [show f = F g by ext; rfl] at hf ⊢
+    exact ⟨g, hf, rfl⟩
+#align bounded_continuous_function.arzela_ascoli₂ BoundedContinuousFunction.arzela_ascoli₂
+
+/-- Third (main) version, with pointwise equicontinuity and range in a compact subset, but
+without closedness. The closure is then compact. -/
+theorem arzela_ascoli [T2Space β] (s : Set β) (hs : IsCompact s) (A : Set (α →ᵇ β))
+    (in_s : ∀ (f : α →ᵇ β) (x : α), f ∈ A → f x ∈ s) (H : Equicontinuous ((↑) : A → α → β)) :
+    IsCompact (closure A) :=
+  /- This version is deduced from the previous one by checking that the closure of `A`, in
+  addition to being closed, still satisfies the properties of compact range and equicontinuity. -/
+  arzela_ascoli₂ s hs (closure A) isClosed_closure
+    (fun _ x hf =>
+      (mem_of_closed' hs.isClosed).2 fun ε ε0 =>
+        let ⟨g, gA, dist_fg⟩ := Metric.mem_closure_iff.1 hf ε ε0
+        ⟨g x, in_s g x gA, lt_of_le_of_lt (dist_coe_le_dist _) dist_fg⟩)
+    (H.closure' continuous_coe)
+#align bounded_continuous_function.arzela_ascoli BoundedContinuousFunction.arzela_ascoli
+
+end ArzelaAscoli
+
+section One
+
+variable [TopologicalSpace α] [PseudoMetricSpace β] [One β]
+
+@[to_additive]
+instance : One (α →ᵇ β) :=
+  ⟨const α 1⟩
+
+@[to_additive (attr := simp)] -- Porting note: Originally `@[simp, to_additive]`
```

Comment:
```suggestion
@[to_additive (attr := simp)]
```
There is no need to put a porting note on this; it's just the standard fix, and we don't have to revisit it afterwards.